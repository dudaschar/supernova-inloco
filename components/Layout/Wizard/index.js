import cx from 'classnames'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

import { Layout } from '../'
import Wizard from '../../Wizard'

export class LayoutWizard extends Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node.isRequired,
    initialStepIndex: PropTypes.number,
    mainContentProps: PropTypes.object,
    onStepChange: PropTypes.func,
    steps: PropTypes.arrayOf(PropTypes.string).isRequired
  }

  static defaultProps = {
    initialStepIndex: 0,
    mainContentProps: {}
  }

  state = {
    currentStep: this.props.initialStepIndex
  }

  render() {
    const {
      children,
      className,
      initialStepIndex,
      mainContentProps,
      steps,
      ...otherProps
    } = this.props
    const { currentStep } = this.state
    const classes = cx('inloco-layout__wizard', className)
    return (
      <React.Fragment>
        <Layout.Header>
          <Layout.HeaderTitle>
            <div className="inloco-layout__wizard-title">
              <div className="inloco-layout__wizard-title-number">
                {currentStep + 1}
              </div>
              {steps[currentStep]}
            </div>
          </Layout.HeaderTitle>
        </Layout.Header>
        <Layout.MainContent {...mainContentProps}>
          <Wizard
            className={classes}
            currentStep={currentStep}
            steps={steps}
            onChangeStep={this.handleChangeStep}
            {...otherProps}>
            {children}
          </Wizard>
        </Layout.MainContent>
      </React.Fragment>
    )
  }

  handleChangeStep = newStep => {
    const { onStepChange } = this.props
    onStepChange && onStepChange(newStep)
    this.setState({ currentStep: newStep })
  }
}

export default LayoutWizard
