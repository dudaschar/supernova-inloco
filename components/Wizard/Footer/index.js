import cx from 'classnames'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

import { Breadcrumb, Icon } from 'semantic-ui-react'

class WizardFooter extends Component {
  static propTypes = {
    className: PropTypes.string,
    controls: PropTypes.func.isRequired,
    initialStepIndex: PropTypes.number,
    onChangeStep: PropTypes.func,
    steps: PropTypes.arrayOf(PropTypes.string).isRequired
  }

  static defaultProps = {
    initialStepIndex: 0,
    onChangeStep: () => {}
  }

  state = {
    currentStep: this.props.initialStepIndex
  }

  render() {
    const {
      controls,
      className,
      initialStepIndex,
      onChangeStep,
      steps,
      ...otherProps
    } = this.props
    const { currentStep } = this.state
    const classes = cx('inloco-wizard__footer', className)
    return (
      <div className={classes} {...otherProps}>
        <Breadcrumb>
          {steps.map((step, index) => {
            const sectionClasses = cx({
              'inloco-wizard__footer-step--current': currentStep === index
            })
            return (
              <React.Fragment key={step}>
                {index !== 0 && (
                  <Breadcrumb.Divider>
                    <Icon className="chevron right" />
                  </Breadcrumb.Divider>
                )}
                <Breadcrumb.Section className={sectionClasses}>
                  {step}
                </Breadcrumb.Section>
              </React.Fragment>
            )
          })}
        </Breadcrumb>
        <div className="inloco-wizard__footer-controls">
          {controls({
            currentStep,
            onChangeStep: this.handleChangeStep
          })}
        </div>
      </div>
    )
  }

  handleChangeStep = newStepIndex => {
    this.setState({ currentStep: newStepIndex })
    this.props.onChangeStep(newStepIndex)
  }
}

export default WizardFooter
