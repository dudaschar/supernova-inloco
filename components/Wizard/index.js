import cx from 'classnames'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

import WizardFooter from './Footer'
import WizardFooterBasicControls from './Footer/BasicControls'

export class Wizard extends Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node.isRequired
  }

  static Footer = WizardFooter
  static FooterBasicControls = WizardFooterBasicControls

  render() {
    const { children, className, ...otherProps } = this.props
    const classes = cx('inloco-wizard', className)
    return (
      <div className={classes} {...otherProps}>
        {children}
      </div>
    )
  }
}

export default Wizard
