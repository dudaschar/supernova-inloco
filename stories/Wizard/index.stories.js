import React from 'react'
import { storiesOf } from '@storybook/react'

import { Button, Wizard } from '../../components'

storiesOf('Wizard', module).add('footer', () => (
  <Wizard.Footer
    controls={props => <Wizard.FooterBasicControls {...props} totalSteps={3} />}
    steps={['First Step', 'Second Step', 'Third Step']}
  />
))
