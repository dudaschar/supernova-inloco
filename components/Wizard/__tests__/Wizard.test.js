import { shallow } from 'enzyme'
import React from 'react'

import Wizard from '../'

const buildWrapper = props => shallow(<Wizard>Test Children Content</Wizard>)

describe('Wizard', () => {
  it('should render wizard with given content', () => {
    expect(buildWrapper()).toMatchSnapshot()
  })
})
