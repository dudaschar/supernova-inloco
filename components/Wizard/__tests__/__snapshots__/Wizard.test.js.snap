// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard should render wizard with given content 1`] = `
<div
  className="inloco-wizard"
>
  Test Children Content
</div>
`;
